dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/WorkspaceItemRestRepository.java

```diff
@@ -0,0 +1,315 @@
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.EPersonServiceImpl;
+import org.dspace.services.ConfigurationService;
+import org.dspace.submit.AbstractProcessingStep;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+/**
+ * This is the repository responsible to manage WorkspaceItem Rest object
+ * 
+ * @author Andrea Bollini (andrea.bollini at 4science.it)
+ *
+ */
+
+@Component(WorkspaceItemRest.CATEGORY + "." + WorkspaceItemRest.NAME)
+public class WorkspaceItemRestRepository extends DSpaceRestRepository<WorkspaceItemRest, Integer> {
+
+	public static final String OPERATION_PATH_SECTIONS = "sections";
+	
+	private static final Logger log = Logger.getLogger(WorkspaceItemRestRepository.class);
+	
+	@Autowired
+	WorkspaceItemService wis;
+	@Autowired
+	ItemService itemService;
+	@Autowired
+	BitstreamService bitstreamService;
+	@Autowired
+	BitstreamFormatService bitstreamFormatService;
+	@Autowired
+	ConfigurationService configurationService;
+	
+	@Autowired
+	WorkspaceItemConverter converter;
+	
+	@Autowired
+	SubmissionService submissionService;
+	@Autowired
+	EPersonServiceImpl epersonService;
+	
+	private SubmissionConfigReader submissionConfigReader;
+	
+	public WorkspaceItemRestRepository() throws SubmissionConfigReaderException {
+		submissionConfigReader = new SubmissionConfigReader();
+	}
+
+	@Override
+	public WorkspaceItemRest findOne(Context context, Integer id) {
+		WorkspaceItem witem = null;
+		try {
+			witem = wis.find(context, id);
+		} catch (SQLException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		}
+		if (witem == null) {
+			return null;
+		}
+		return converter.fromModel(witem);
+	}
+
+	@Override
+	public Page<WorkspaceItemRest> findAll(Context context, Pageable pageable) {
+		List<WorkspaceItem> witems = null;
+		int total = 0;
+		try {
+			total = wis.countTotal(context);
+			witems = wis.findAll(context, pageable.getPageSize(), pageable.getOffset());
+		} catch (SQLException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		}
+		Page<WorkspaceItemRest> page = new PageImpl<WorkspaceItem>(witems, pageable, total).map(converter);
+		return page;
+	}
+	
+	@SearchRestMethod(name = "findBySubmitter")
+	public Page<WorkspaceItemRest> findBySubmitter(@Param(value="uuid") UUID submitterID, Pageable pageable) {
+		List<WorkspaceItem> witems = null;
+		int total = 0;
+		try {		
+			Context context = obtainContext();
+			EPerson ep = epersonService.find(context, submitterID); 
+			witems = wis.findByEPerson(context, ep);
+			total = witems.size();
+		} catch (SQLException e) {
+			throw new RuntimeException(e.getMessage(), e);
```

Comment:
I would just let the SQLException pass and let the controller advice translate it to an error code (see comments above).

Reply:
As above, I prefer to manage exception in other PR.